x/evm/keeper/hooks_test.go

```diff
@@ -0,0 +1,72 @@
+package keeper_test
+
+import (
+	"errors"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+
+	ethermint "github.com/tharsis/ethermint/types"
+	"github.com/tharsis/ethermint/x/evm/keeper"
+	"github.com/tharsis/ethermint/x/evm/types"
+)
+
+// LogRecordHook records all the logs
+type LogRecordHook struct {
+	Logs []*ethtypes.Log
+}
+
+func (dh *LogRecordHook) PostTxProcessing(ctx sdk.Context, txHash common.Hash, logs []*ethtypes.Log) error {
+	dh.Logs = logs
+	return nil
+}
+
+// FailureHook always fail
+type FailureHook struct{}
+
+func (dh FailureHook) PostTxProcessing(ctx sdk.Context, txHash common.Hash, logs []*ethtypes.Log) error {
+	return errors.New("post tx processing failed")
+}
+func (suite *KeeperTestSuite) TestEvmHooks() {
+	suite.SetupTest()
+	suite.Commit()
+
+	logRecordHook := LogRecordHook{}
+	suite.app.EvmKeeper.SetHooks(keeper.NewMultiEvmHooks(&logRecordHook))
+
+	k := suite.app.EvmKeeper
+
+	txHash := common.BigToHash(big.NewInt(1))
+
+	amt := sdk.Coins{ethermint.NewPhotonCoinInt64(100)}
+	err := suite.app.BankKeeper.MintCoins(suite.ctx, types.ModuleName, amt)
+	suite.Require().NoError(err)
+	err = suite.app.BankKeeper.SendCoinsFromModuleToAccount(suite.ctx, types.ModuleName, suite.address.Bytes(), amt)
+	suite.Require().NoError(err)
+
+	k.SetTxHashTransient(txHash)
+	k.AddLog(&ethtypes.Log{
+		Topics:  []common.Hash{},
+		Address: suite.address,
+	})
+
+	logs := k.GetTxLogs(txHash)
+	suite.Require().Equal(1, len(logs))
+
+	err = k.PostTxProcessing(txHash, logs)
+	suite.Require().NoError(err)
+
+	suite.Require().Equal(1, len(logRecordHook.Logs))
+}
+
+func (suite *KeeperTestSuite) TestHookFailure() {
```

Comment:
can we use Table Tests instead of the 2 test functions? See this [code](https://github.com/tharsis/ethermint/pull/515/files#diff-86a2788c6f900e3e8d070486fa1b357f309c85c581bb1b7b64597b7d7445cc8dR19) for reference

Reply:
done